src/extension/declarative/ops.rs

```diff
@@ -0,0 +1,115 @@
+//! Declarative operation definitions.
+//!
+//! This module defines a YAML schema for defining operations in a declarative way.
+//!
+//! See the [specification] and [`ExtensionSetDeclaration`] for more details.
+//!
+//! [specification]: https://github.com/CQCL/hugr/blob/main/specification/hugr.md#declarative-format
+//! [`ExtensionSetDeclaration`]: super::ExtensionSetDeclaration
+
+use std::collections::HashMap;
+use std::path::PathBuf;
+
+use serde::{Deserialize, Serialize};
+use smol_str::SmolStr;
+
+use crate::extension::{ExtensionRegistry, ExtensionSet, OpDef, SignatureFunc};
+use crate::types::type_param::TypeParam;
+use crate::Extension;
+
+use super::signature::SignatureDeclaration;
+use super::ExtensionDeclarationError;
+
+/// A declarative operation definition.
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub(super) struct OperationDeclaration {
+    /// The identifier the operation.
+    name: SmolStr,
```

Comment:
`OpName` like `TypeName`?

Reply:
`OpName` is a trait :/

I wanted to give wrappers for op, type, and parameter names, and stop using (smol)stringly-typed names everywhere.

But that's out of the scope of this PR, so I'll open an issue.